crates/red_knot_python_semantic/resources/mdtest/narrow/conditionals/elif_else.md

```diff
@@ -55,3 +55,16 @@ else:
     # TODO should be Never
     reveal_type(x)  # revealed: Literal[1, 2]
 ```
+
+## After if-statement without explicit else branch, assume else branch with negative clause
```

Comment:
IMHO, need also test for `x` is not changed in the `if` branch like:
```python
def optional_int() -> int | None: ...

x = optional_int()

if x is None: pass

reveal_type(x)  # revealed: int | None
```

Reply:
Good idea. I didn't see it being tested else where. I added another variable in the same mdtest here. I wonder if a new mdtest file would be better to test `post if block` narrowing scenarios.

Reply:
> I wonder if a new mdtest file would be better to test `post if block` narrowing scenarios.

I think that's a good idea. We should also add tests for scenarios with (multiple) `elif` branches, but no `else` branch.

Reply:
I added a new test file and added couple of more tests.